// Copyright 2025 Flower Labs GmbH. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ==============================================================================

syntax = "proto3";

package flwr.proto;

import "flwr/proto/message.proto";
import "flwr/proto/fab.proto";
import "flwr/proto/run.proto";

// PushMessages messages
message PushInsMessagesRequest {
  repeated Message messages_list = 1;
  uint64 run_id = 2;
  repeated ObjectTree message_object_trees = 3;
}
message PushInsMessagesResponse {
  repeated string message_ids = 1;
  map<string, ObjectIDs> objects_to_push = 2;
}

// PullMessages messages
message PullResMessagesRequest {
  repeated string message_ids = 1;
  uint64 run_id = 2;
}
message PullResMessagesResponse {
  repeated Message messages_list = 1;
  map<string, ObjectIDs> objects_to_pull = 2;
}
// PullServerAppInputs messages
message PullServerAppInputsRequest {}
message PullServerAppInputsResponse {
  Context context = 1;
  Run run = 2;
  Fab fab = 3;
}

// PushServerAppOutputs messages
message PushServerAppOutputsRequest {
  uint64 run_id = 1;
  Context context = 2;
}
message PushServerAppOutputsResponse {}
